### ISSUE TITLE
Compatible with Ember 2.12 ?

### ISSUE THREAD

**Issue by TimvdEijnden** [ember-redux/ember-redux#105]
I'm really thrilled to start using Redux in a new Ember project. I'm trying to re-create the number-count twiddle: https://ember-twiddle.com/2d98cd4418b7df5cbce6c5213351d31e

But I keep running into this error:

```
Error: Could not find module `redux` imported from `redux-test/services/redux`
```

Steps to reproduce:

- clone this repo: https://github.com/TimvdEijnden/ember-redux-number-count and do yarn install:

Or

- npm install ember-cli@2.12 -g
- ember new number-count
- cd number-count
- ember install ember-redux
- ember g component number-count
- add the following to components/number-count.js
```js
import Ember from 'ember';
import connect from 'ember-redux/components/connect';

var stateToComputed = (state) => {
  return {
    number: state.number
  };
};

var dispatchToActions = (dispatch) => {
  return {
    add: () => dispatch({type: 'ADD'})
  };
};

var NumbersComponent = Ember.Component.extend({});

export default connect(stateToComputed, dispatchToActions)(NumbersComponent);

```

- add the following to templates/components/number-count.hbs

 ```hbs
{{number}}
<button onclick={{action "add"}}>add</button>
```

- add the following to reducers/index.js

```js
var number = ((state, action) => {
    if(action.type === 'ADD') {
        return state + 1;
    }
    return state || 0;
});

export default {
    number
}
```
- add the `{{number-count}}` to the application.hbs
- ember s

Now the error occurs.

**Comment by toranb** [comment]
Great catch! I'm curious if this is an ember-cli v2.12 issue (not ember v3.12 itself). I was using an early beta of ember v2.12 no problem but havent yet looked at upgrading ember-cli

**Comment by TimvdEijnden** [comment]
Maybe it has to do something with ember.js moving from bower to the npm module ember-source?

**Comment by rwwagner90** [comment]
@toranb I downgraded from ember-cli 2.12 to 2.11 and the error changed from could not find redux to:

`TypeError: null is not an object (evaluating 'dataSources.includes')`

**Comment by toranb** [comment]
Wow not good - I'll need to dive in and likely chat with Stef or Robert on this one (sorry guys)

**Comment by rwwagner90** [comment]
Looks like it's related to `treeForAddon` in the shims. Thanks to @dbashford for finding this issue! https://github.com/ember-cli/ember-cli/issues/6826

**Comment by toranb** [comment]
@TimvdEijnden looks like ember-cli v2.12 is not so friendly at the moment. Our good friend @rwwagner90 did create a new version of these shim addons that works as shown below

https://github.com/toranb/ember-redux-with-cli-twelv/commit/e5245660d366ab60ad0aab197dc0125cc5f6f4b0

an example app built with ember-cli v2.12 and ember-redux (the new shims) ^^

The problem we face now is that the new shims used to fix ember-cli v2.12 don't work for something older like ember-cli v2.8 (in my experience)

I think we need some time yet to work out if we can support both old/new ember-cli alike with `ember-cli-node-assets` ... or if not we may need to look at only supporting ember-cli 2.9+ or something else (as I don't want to revert back to `ember-browserify` if we can help it). Any other options worth considering - please let me know and sorry for the delay but I'd prefer to slow down and look at all the options if we can :)

**Comment by TimvdEijnden** [comment]
Thanks for the update, no rush! With the quick fix 👍  by @rwwagner90 we can move forward!

**Comment by rwjblue** [comment]
Seems like the same issue as https://github.com/mike-north/ember-lodash/issues/89. Which was fixed by https://github.com/mike-north/ember-lodash/pull/93. 

tldr; we make some optimizations when the addon directory does not contain any *.js files (which is also true of ember-redux-shim ), and avoid doing needless work. The solution is to add the following to ember-redux-shim's `index.js`:

```
  _shouldCompileJS: function() {
    return true;
  },
```

**Comment by TimvdEijnden** [comment]
Ember-cli 2.12.1 has been released which fixes this issue: https://github.com/ember-cli/ember-cli/releases/tag/v2.12.1

**Comment by toranb** [comment]
Awesome! If all is well I'll close this issue 🙂

Thanks for reporting this so quickly! And thanks to @rwjblue for fixing it so quickly!
